joern-cli/frontends/kotlin2cpg/src/main/scala/io/joern/kotlin2cpg/types/DefaultTypeInfoProvider.scala

```diff
@@ -706,28 +608,24 @@ class DefaultTypeInfoProvider(environment: KotlinCoreEnvironment) extends TypeIn
       if (fnDesc.isDefined && fnDesc.get.getExtensionReceiverParameter != null) {
         val erpType = fnDesc.get.getExtensionReceiverParameter.getType
         if (erpType.isInstanceOf[UnresolvedType]) {
-          TypeConstants.cpgUnresolved + "." + expr.getName
+          s"${TypeConstants.cpgUnresolved}.${expr.getName}"
         } else {
           val theType      = fnDesc.get.getExtensionReceiverParameter.getType
           val renderedType = TypeRenderer.render(theType)
-          renderedType + "." + expr.getName
+          s"$renderedType.${expr.getName}"
         }
-      } else {
-        expr.getFqName
-      }
+      } else expr.getFqName
 
-    val signature = returnTypeFullName + paramListSignature
+    val signature = s"$returnTypeFullName$paramListSignature"
     val fullname  = s"$methodName:$signature"
     (fullname, signature)
   }
 
   def isReferenceToClass(expr: KtNameReferenceExpression): Boolean = {
     descriptorForNameReference(expr)
       .collect {
-        case _: LazyJavaClassDescriptor =>
-          true
-        case _: LazyClassDescriptor =>
-          true
+        case _: LazyJavaClassDescriptor => true
+        case _: LazyClassDescriptor     => true
       }
       .getOrElse(false)
   }
```

Comment:
This could be replaced by an `exists` call with a catch-all third case (see my comment about `isConstructorCall`). It's less of a thing here since this already returns a `Boolean`, however.

Reply:
:+1: